src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -0,0 +1,491 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // Contains test methods that can be used for FileInfo, DirectoryInfo, File or Directory.
+    public abstract class BaseSymbolicLinks_FileSystem : BaseSymbolicLinks
+    {
+        /// <summary>Creates a new file or directory depending on the implementing class.
+        /// If createOpposite is true, creates a directory if the implementing class is for File or FileInfo, or
+        /// creates a file if the implementing class is for Directory or DirectoryInfo.</summary>
+        protected abstract void CreateFileOrDirectory(string path, bool createOpposite = false);
+
+        protected abstract void AssertIsCorrectTypeAndDirectoryAttribute(FileSystemInfo linkInfo);
+
+        protected abstract void AssertLinkExists(FileSystemInfo linkInfo);
+
+        /// <summary>Calls the actual public API for creating a symbolic link.</summary>
+        protected abstract FileSystemInfo CreateSymbolicLink(string path, string pathToTarget);
+
+        private void CreateSymbolicLink_Opposite(string path, string pathToTarget)
+        {
+            Type t = GetType();
+            if (t == typeof(Directory_SymbolicLinks) || t == typeof(DirectoryInfo_SymbolicLinks))
```

Comment:
this check might cause a surprise if we ever try to add more types that derive from `BaseSymbolicLinks_FileSystem`

perhaps we could introduce an abstract property like:

```cs
protected abstract bool IsDirectoryTest { get; }
```

and just use it instead?